src/hayhooks/server/utils/mcp_availability.py

```diff
@@ -0,0 +1,40 @@
+import importlib.util
+import functools
+import sys
+from hayhooks import log
+
+MCP_AVAILABLE = importlib.util.find_spec("mcp") is not None
+
+MCP_ERROR_MSG = (
+    "MCP functionality requires the 'mcp' package. "
+    "Ensure you're running Python 3.10 or higher. "
+    "Then install with 'pip install hayhooks[mcp]'"
+)
+
+
+def ensure_mcp_available():
+    """Raise an informative error if MCP is not available"""
+    if not MCP_AVAILABLE:
+        log.error(MCP_ERROR_MSG)
+        raise ImportError(MCP_ERROR_MSG)
+
+
```

Comment:
Have you considered using Haystack `LazyImport` (`DeferredImportExceptionContextManager`)?
https://github.com/deepset-ai/haystack/blob/67de0369c711c7e3440c5b956a4b090820747abd/haystack/lazy_imports.py#L13

Reply:
This is interesting since it could make the code more readable, I'll do a try ;)

Reply:
I am using `LazyImport` in code now, it seems to work and it's cleaner, thanks for the nice suggestion!

However, I've kept `MCP_AVAILABLE = importlib.util.find_spec("mcp") is not None` in `test_mcp` since I want a boolean to skip test if package is not installed.